lib/pico-sdk: Update pico-sdk to 2.2.0 and define PIOASM_VERSION_STRING.

Bohrium332 opened the pull request:
### Summary

Update the pico-sdk submodule to bring in Raspberry Pi Pico SDK v2.2.0 (updating from v2.0.0).

New feature support from upstream Pico SDK v2.2.0: enhanced ADC/DAC functionality, improved SPI/I2C peripheral stability, and expanded board definition coverage
Compatibility updates for RP2350-based devices with critical hardware macro definitions
No breaking changes to existing RP2040/RP2350 device support or MicroPython RP2 port APIs


### Testing

The rebase completed successfully with no conflicts. All MicroPython-specific patches applied cleanly.
This update has been validated with:
Compilation tests for existing RP2040/RP2350 board targets (e.g.Seeed XIAO RP2350)
Functional verification of core peripherals (ADC, SPI, I2C, UART) on Seeed XIAO RP2350
Compatibility checks with the latest MicroPython RP2 port codebase


### Trade-offs and Alternatives

This is a vendor library update that integrates bug fixes, performance improvements, and hardware support enhancements from the upstream Raspberry Pi Pico SDK. The submodule points to the work branch containing all MicroPython customizations, ensuring:
Full compatibility with the existing MicroPython RP2 port

Bohrium332 committed:
lib/pico-sdk: Update pico-sdk to 2.2.0 and define PIOASM_VERSION_STRING.

Signed-off-by: zhida zhang<zhida.zhang@seeed.cc>

dpgeorge commented:
@Bohrium332 please do not keep opening and closing PRs.  Ask for help if you need it.

Bohrium332 commented:
> @Bohrium332请不要频繁地打开和关闭 PR。如果需要帮助，请寻求帮助。

Hello I need your help yesterday I submit a pr:https://github.com/micropython/micropython/pull/18444, I changed a place that doesn't matter, when robot review several robot errors:

windows port / build-vs (x86, Release, standard, 2019) (pull_request)

windows port / build-vs (x64, Release, dev, 2017) (pull_request)

windows port / build-vs (x64, Release, dev, 2019) (pull_request)

windows port / build-vs (x86, Release, dev, 2017) (pull_request)

The specific error message is as follows:

```
Download of Catalog_1222827674.man (7.34MB) completed.

Downloading channel manifest

from 'https://aka.ms/vs/17/release/channel'

WARNING:  C:\Users\runneradmin\AppData\Local\Temp\chocolatey\chocolatey-visualstudio.extension\ChannelManifest_574806083.man is of  content type text/html;  charset=utf-8



Download of ChannelManifest_574806083.man (-1 B) completed.

ERROR: Exception calling "DeserializeObject" with "1" argument(s): "Invalid JSON primitive: ."

The install of visualstudio2017buildtools was NOT successful.

Error while running 'C:\ProgramData\chocolatey\lib\visualstudio2017buildtools\tools\ChocolateyInstall.ps1'.

See log for details.



Chocolatey installed 0/1 packages. 1 packages failed.

See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).



Failures

- visualstudio2017buildtools (exited -1) - Error while running  'C:\ProgramData\chocolatey\lib\visualstudio2017buildtools\tools\ChocolateyInstall.ps1'.

See log for details.



Enjoy using Chocolatey?  Explore more amazing features to take your

experience to the next level at

https://chocolatey.org/compare

Error: Process completed with exit code 127.
```

Then I thought maybe there was a problem with what I modified. So I opened a new PR and resubmitted what was successfully reviewed by the robot yesterday (that is, this pr), but it also reported an error. The contents of the two modifications were exactly the same. Could you please help me check what the problem was? Thank you

Bohrium332 commented:
@dpgeorge 
I'm sorry I shouldn't keep turning PR open and close.
May I ask if you can see the log of the robot's inspection.
My PR didn't have these failed checks yesterday.

dpgeorge commented:
The CI failures are unrelated to your changes, so you can ignore them.  I have rerun them to try and get them to pass.

(We have a lot of CI in this project, and the CI can break down for unrelated reasons.)

Bohrium332 committed:
rp2: Remove stack size setting for Seeed XIAO RP2350.

Signed-off-by: zhida zhang <zhida.zhang@seeed.cc>

Bohrium332 commented:
> The CI failures are unrelated to your changes, so you can ignore them. I have rerun them to try and get them to pass.
> 
> (We have a lot of CI in this project, and the CI can break down for unrelated reasons.)

I have completed all my changes. Could you please return the failed ci. Thank you very much.

lurch commented:
For cross-reference purposes: this PR originates from a discussion in https://github.com/raspberrypi/pico-sdk/issues/2718

Bohrium332 commented:
@dpgeorge This robot seems to be stuck

lurch commented:
#18465 has been approved, so perhaps this PR should be closed?
